src/tool/hpcrun/gpu/nvidia/cubin-symbols.c

```diff
@@ -221,12 +243,24 @@ relocateSymbols
   GElf_Ehdr ehdr_v;
   GElf_Ehdr *ehdr = gelf_getehdr(elf, &ehdr_v);
   if (ehdr) {
+    // If .symtab_shndx section exists, the section index of a symbol
```

Comment:
I think that it would be better to have a class that hides whether there is a SHT_SYMTAB_SHNDX section internally.

The class constructor could include this loop to initialize its symtab_shndx_section field.